packages/jmix-react-antd/src/ui/EntityHierarchyTree/EntityHierachyTree.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import {Tree, TreeProps} from "antd";
+import {DataNode} from "rc-tree/lib/interface";
+
+export type HierarchyTreeNode = Omit<DataNode, "children" | "key"> & {
+  id: string,
+  _instanceName: string,
+  [child: string]: string,
+};
+
+export type EntityHierarchyTreeProps = Omit<TreeProps, "treeData"> & {
+  hierarchyProperty: string,
+  items?: HierarchyTreeNode[],
```

Comment:
The component should receive an array of entity instances, i.e. type `EntityInstance<unknown, WithId, WithInstanceName>[]`.

Reply:
Type `WithId` defines prop id as `EntityId` what means it can be `string | object | number`, while antd `Tree` prop `treeData` expect unique keys for each `DataNode`, that can be only `string | number`. Because of differences in `id` and `key` types i can't directly use one instead of other, i need to implement some match map for `id - (generated) key` and use it every time when `key` is used in `Tree`, is it worth doing so or may be there is another way?

Reply:
I think it worth doing so. Framework user should not be concerned by these implementation details. They should be able to just feed their `EntityInstance` array to the component. The mapping of ids should happen inside `EntityHierarchyTree` component. I think it should be trivial to map the ids by using our `toIdString()` util function.